src/controllers/userController.js

```diff
@@ -0,0 +1,64 @@
+const User = require("../models/user")
+
+
+
+//Fetch User
+function fetchUsers(req, res){
+    User.find({}, (err, users) => {
+      if(err) {
+          return res.status(500).json({ message: err})
+      }else {
+          return res.status(200).json({ users })
+      }
+  })
+}
+  
+  
+
+
+//Fetch Single User
+ function fetchUserById(req, res){
+  let id = req.params.id
+  User.findById(id, (err, user) => {
+    if(err) {
+      return res.status(500).json({ message: err})
+  }
+  else if(!user){
+      return res.status(404).json({ message: "user not found"})
+  }
+  else {
+      return res.status(200).json({ user })
+  }
+
+  })
+}
+
+
+function fetchUsersProfile(req, res){
+    let { id } = req.user
+    let currentUser = {
+        firstName: req.user.firstName,
+        lastName: req.user.lastName,
+        email: req.body.email,
+        role: req.body.role
+    }
+   
+    User.findById(id, (err, user) => {
+        if(err) {
+          return res.status(500).json({ message: err})
+      }
+      else if(!user){
+          return res.status(404).json({ message: "user not found"})
+      }
+      else {
+          return res.status(200).json({ currentUser })
```

Comment:
so what you should be returning here should be the `user` you just found

Reply:
Yeah I was trying to return the user without the password. I just noticed that Dorian added the email and role properties in the jwt in last PR. So I will just delete the currentUser then.